tests/datatypes/test_dict.py

```diff
@@ -229,6 +229,66 @@ def test_method_get(self):
                 print(err)
             """)
 
+    def test_copy(self):
+        self.assertCodeExecution("""
+            x = {42: 'Babel'}
+            y = x.copy()
+            print(y)
+            """)
+
+        self.assertCodeExecution("""
+            x = {42: 'Babel'}
+            y = x.copy()
+            print(x == y)
+            """)
+
+        self.assertCodeExecution("""
+            x = {42: 'Babel'}
+            y = x.copy()
+            print(x is not y)
+            """)
+
+        self.assertCodeExecution("""
+            x = {42: 'Babel'}
+            y = x.copy()
+            y['42'] = 'Babel'
+            print(x == y)
+            """)
+
+        self.assertCodeExecution("""
+            x = {'key': {42: 'Babel'}, 42: 'Babel'}
+            y = x.copy()
+            print(x['key'] is y['key'])
+            """)
+
+    def test_fromkeys(self):
+        self.assertCodeExecution("""
+            keys = [1, 2]
+            print({}.fromkeys(keys))
+            """)
+
+        # non-iterable error test
+        self.assertCodeExecution("""
+            keys = [1, 2]
+            value = 2
+            print({}.fromkeys(keys, value))
+            """)
+
+        self.assertCodeExecution("""
+            keys = (1, 2)
+            value = 5
+            print({}.fromkeys(keys, 5))
+            """)
+
+        # non-hashable error on key test
+        self.assertCodeExecution("""
+            keys = [[1], 2]
+            try:
+                print({}.fromkeys(keys))
+            except TypeError as err:
+                print(err)
+            """)
```

Comment:
Just two small things – could you add a few lines to ensure that `dict.fromkeys()` doesn't affect an existing dict, and also that it can handle empty iterables?